File: MARCH 24/08 March/Check if frequencies can be equal gfg.cpp
class Solution{
public:	
	bool sameFreq(string s)
	{
	    // code here 
        vector<int> freq(26,0);
	    for(char ch:s)
	        freq[ch-'a']++;
	    int maxi = *max_element(freq.begin(), freq.end());
	    
	    int val =0 , cnt=0, cnt1=0;
	    
	    for(int i=0; i<26; i++)
	    {
	        if(freq[i]!=0)
	        {
	            if(freq[i]<maxi)
	            {
	                cnt1++;
	                if(val==0)
	                    val = freq[i];
	                else if(freq[i]!=val)
	                    return false;
	            }
	            else 
	                cnt++;
	        }
	    }
	    
	   // cout<<cnt<<" "<<cnt1<<endl;
	   // cout<<maxi<<" "<<val<<endl;
	    
	    if((cnt>1 && cnt1>1) || (cnt==1 && cnt1>1 && maxi-val>1) || (cnt1==1 && cnt>1 && val!=1) || (cnt1==1 && cnt==1 && maxi-val>1))
	        return false;
	    return true;
	}
};

File: JUNE 2025/25 June/Check if frequencies can be equal gfg.java
class Solution {
    boolean sameFreq(String s) {
        // code here
        int freq[]=new int[26];
        for(char i:s.toCharArray()){
            freq[i-'a']++;
        }
        HashMap<Integer,Integer> map=new HashMap<>();
        int a=-1,b=-1;
        for(int i=0;i<26;i++){
        if(freq[i]!=0){
            if(map.get(freq[i])!=null)map.put(freq[i],map.get(freq[i])+1);
            else map.put(freq[i],1);
         if(a==-1) a=freq[i];
         else if(b==-1 && a!=freq[i]) b=freq[i];}
        }
        if(map.size()>2) return false;
        if(map.size()<2) return true;
        if(a==-1 || b==-1 || a==1 || b==1) return true;
        if(Math.abs(a-b)==1 && map.get(Math.max(a,b))==1) return true;
        return false;
    
    }
}
